app/com/arpnetworking/kairos/client/models/MetricsQuery.java

```diff
@@ -0,0 +1,547 @@
+/**
+ * Copyright 2017 Smartsheet
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.kairos.client.models;
+
+import com.arpnetworking.commons.builder.OvalBuilder;
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.LinkedHashMultimap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Multimap;
+import net.sf.oval.constraint.Min;
+import net.sf.oval.constraint.NotEmpty;
+import net.sf.oval.constraint.NotNull;
+import org.joda.time.DateTime;
+import org.joda.time.Period;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Model class to represent a metrics query.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public final class MetricsQuery {
+    /**
+     * Gets the start time of the query in milliseconds.
+     *
+     * @return the start time in milliseconds
+     */
+    @JsonProperty("start_absolute")
+    public long startMillis() {
+        return _startTime.getMillis();
+    }
+
+    /**
+     * Gets the end time of the query in milliseconds.
+     *
+     * @return the end time in milliseconds
+     */
+    @JsonProperty("end_absolute")
+    public long endMillis() {
+        return _endTime.getMillis();
+    }
+
+    @JsonProperty("metrics")
+    public List<Metric> getMetrics() {
+        return _metrics;
+    }
+
+    private MetricsQuery(final Builder builder) {
+        _startTime = builder._startTime;
+        _endTime = builder._endTime;
+        _metrics = ImmutableList.copyOf(builder._metrics);
+    }
+
+    private final DateTime _startTime;
+    private final DateTime _endTime;
+    private final List<Metric> _metrics;
```

Comment:
Why not use `ImmutableList` in the builder to avoid copies?

Reply:
Sure. Done